src/parse/matches/arg_matches.rs

```diff
@@ -234,14 +234,31 @@ impl ArgMatches {
             fn to_str_slice(o: &OsString) -> &str {
                 o.to_str().expect(INVALID_UTF8)
             }
-            let to_str_slice: fn(&OsString) -> &str = to_str_slice; // coerce to fn pointer
 
             Values {
-                iter: arg.vals.iter().map(to_str_slice),
+                iter: arg.vals_flatten().map(to_str_slice),
             }
         })
     }
 
+    /// Placeholder documentation.
+    pub fn grouped_values_of<T: Key>(&self, id: T) -> Option<GroupedValues> {
```

Comment:
From a design point, do we need this function? Or can we get away with combining this into `values_of` so that users don't have to call a different function?

Unless for an arg with multiple values, we want both `grouped_values_of` and `values_of` providing different outputs.

Reply:
Combining this function into `values_of` may introduce additional overhead to people that doesn't care about value grouping. And also, I think most people don't care it even when they are using `multiple`. So my suggestion is providing a function beside `values_of` for the prople who cares about the value grouping.